homeassistant/components/fan/alfawise.py

```diff
@@ -0,0 +1,121 @@
+"""
+Alfawise fan platform (fan is used as a humidifier).
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/fan.alfawise/
+"""
+import logging
+import ipaddress
+from typing import Any
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.fan import (SPEED_LOW, SPEED_HIGH,
+                                          FanEntity, SUPPORT_SET_SPEED)
+from homeassistant.const import CONF_DEVICES, CONF_NAME, CONF_MAC
+from homeassistant.const import STATE_OFF
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.exceptions import PlatformNotReady
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pyAlfawise==0.5-beta']
+
+
+def isip(value: Any) -> str:
+    """Validate that the value is an valid ip."""
+    if value is None:
+        raise vol.Invalid('an empty string is not a valid ip address')
+
+    try:
+        ipaddress.ip_address(value)
+    except ValueError:
+        raise vol.Invalid('not a valid ip address')
+
+    return value
+
+
+DEVICE_SCHEMA = vol.Schema({
+    vol.Required(CONF_MAC): cv.string,
+    vol.Required(CONF_NAME): cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {vol.Required(CONF_DEVICES): {isip: DEVICE_SCHEMA}})
```

Comment:
It shouldn't be limited to IP addresses. The devices could have a DNS name depending on the configuration.

Reply:
This is a limit of the third party library. IP must be provided.